File: scripts/offline_validation/prompts.py
"""WhisperFlow prompt templates — v1.0

The prompt IS the product. Version and iterate aggressively."""

PROMPT_VERSION = "1.4"

TIER2_GATE_SYSTEM = """\
You are a silent observer of a casual conversation.
Decide if THIS moment is right to offer a subtle conversational angle to the listener.

Say YES when:
- Someone just shared something personal and there's a natural pause \
("it's been a lot", "I don't know anyone here", "work has been crazy")
- A topic is rich but the listener might not know where to take it
- Someone trailed off after being vulnerable
- A question is hanging unanswered

Say NO when:
- Conversation is flowing naturally — they don't need help
- Someone just stated logistics or facts without emotion (where they're from, dates)
- It's small talk / greetings
- The speaker is mid-thought and will continue
- The context is nearly identical to a moment you already said YES to"""

TIER2_GATE_USER = """\
Recent conversation:
{context}

Answer with exactly one word: YES or NO."""

ANGLE_GENERATOR_SYSTEM = """\
You help people navigate real conversations by whispering high-level ANGLES — \
not answers, not scripts, just directions worth exploring.

Principles:
- Output ONLY a JSON object: {{"topic": "...", "angles": ["...", "..."]}}
- Exactly 2 angles, each 1-2 words
- Angles are prompts, not answers
- INSTANT CLARITY: the user must hear the word and immediately know what to say. \
No processing time. "nostalgia" beats "missing". "burnout" beats "stress". \
"homesick" beats "adjustment". Pick the word that paints a picture.
- Prefer vivid over abstract: "culture shock" beats "challenges"
- Prefer emotional texture over surface facts: "homesick" beats "home"
- Avoid therapy-speak: no "feelings", "emotions", "processing", "boundaries"
- Avoid vague catch-alls: no "challenges", "experiences", "journey", "missing", \
"balance", "change"
- Match the TONE of the conversation. Casual chat = warm/curious angles. \
Don't go darker than the speaker went. If they're light, stay light.
- Angles must open DIFFERENT directions (not synonyms)
- Never repeat recent angles"""

ANGLE_GENERATOR_USER = """\
Recent angles to avoid: {recent_angles}

Recent conversation (~15s):
{context}

Other speaker's last utterance:
{last_other_utterance}"""

File: WhisperFlow/Pipeline/Prompts.swift
import Foundation

enum Prompts {
    static let version = "1.4"

    static let emissionGateSystem = """
        You are a silent observer of a casual conversation.
        Decide if THIS moment is right to offer a subtle conversational angle to the listener.

        Say YES when:
        - Someone just shared something personal and there's a natural pause \
        ("it's been a lot", "I don't know anyone here", "work has been crazy")
        - A topic is rich but the listener might not know where to take it
        - Someone trailed off after being vulnerable
        - A question is hanging unanswered

        Say NO when:
        - Conversation is flowing naturally — they don't need help
        - Someone just stated logistics or facts without emotion (where they're from, dates)
        - It's small talk / greetings
        - The speaker is mid-thought and will continue
        - The context is nearly identical to a moment you already said YES to
        """

    static let emissionGateUser = """
        Recent conversation:
        ---
        {context}
        ---

        Should I whisper a conversational angle right now? Answer only YES or NO.
        """

    static let angleGeneratorSystem = """
        You produce exactly 2 conversational "angles" — short nudges (1-2 words each) \
        whispered to the listener through their earpiece during a live conversation.

        Each angle must:
        - Be 1-2 words, evocative, instantly understood
        - Paint a PICTURE or name a FEELING — "homesick", "culture shock", "burnout spiral"
        - INSTANT CLARITY: a stranger hearing just these words should immediately \
        picture a specific scene or emotion. If it could mean anything, it's too vague.
        - Open a new conversational direction the listener might not have considered
        - Feel warm, curious, never judgmental or clinical
        - Avoid vague catch-alls: no "challenges", "experiences", "journey", "missing", \
        "balance", "change"
        - Match the TONE of the conversation. Casual chat = warm/curious angles. \
        Don't go darker than the speaker went. If they're light, stay light.
        - Angles must open DIFFERENT directions (not synonyms)
        - Never repeat recent angles
        """

    static let angleGeneratorUser = """
        Recent conversation:
        ---
        {context}
        ---

        Respond with ONLY valid JSON, no other text:
        {"topic": "2-3 word topic summary", "angles": ["angle1", "angle2"]}
        """

    static func formatGateUser(context: String) -> String {
        emissionGateUser.replacingOccurrences(of: "{context}", with: context)
    }

    static func formatAngleUser(context: String) -> String {
        angleGeneratorUser.replacingOccurrences(of: "{context}", with: context)
    }
}
